x-pack/plugin/spatial/src/test/java/org/elasticsearch/xpack/spatial/index/mapper/PointFieldMapperTests.java

```diff
@@ -0,0 +1,279 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+package org.elasticsearch.xpack.spatial.index.mapper;
+
+import org.apache.lucene.util.BytesRef;
+import org.elasticsearch.common.bytes.BytesReference;
+import org.elasticsearch.common.xcontent.XContentBuilder;
+import org.elasticsearch.common.xcontent.XContentFactory;
+import org.elasticsearch.common.xcontent.XContentType;
+import org.elasticsearch.index.mapper.DocumentMapper;
+import org.elasticsearch.index.mapper.Mapper;
+import org.elasticsearch.index.mapper.ParsedDocument;
+import org.elasticsearch.index.mapper.SourceToParse;
+import org.elasticsearch.xpack.spatial.common.CartesianPoint;
+
+import java.io.IOException;
+
+import static org.hamcrest.Matchers.equalTo;
+import static org.hamcrest.Matchers.instanceOf;
+import static org.hamcrest.Matchers.not;
+import static org.hamcrest.Matchers.notNullValue;
+
+public class PointFieldMapperTests extends CartesianFieldMapperTests {
+
+    @Override
+    protected String getFieldName() {
+        return "point";
+    }
+
+    @Override
+    protected void registerParameters(ParameterChecker checker) throws IOException {
+        checker.registerConflictCheck("doc_values", b -> b.field("doc_values", false));
+        checker.registerConflictCheck("index", b -> b.field("index", false));
+        checker.registerUpdateCheck(b -> b.field("ignore_malformed", true), m -> {
+            PointFieldMapper gpfm = (PointFieldMapper) m;
+            assertTrue(gpfm.ignoreMalformed());
+        });
+        checker.registerUpdateCheck(b -> b.field("ignore_z_value", false), m -> {
+            PointFieldMapper gpfm = (PointFieldMapper) m;
+            assertFalse(gpfm.ignoreZValue());
+        });
+    }
+
+    public void testValuesStored() throws Exception {
+        DocumentMapper mapper = createDocumentMapper(fieldMapping(b -> {
+            b.field("type", "point");
+            b.field("store", true);
+        }));
+
+        SourceToParse sourceToParse = new SourceToParse("test", "1",
```

Comment:
You can also use the `source` method, which simplifies this a bit:
```
SourceToParse sourceToParser = source(b -> {
    b.startObject(FIELD_NAME).field("x", 2000.1).field("y", 305.6).endObject();
});
```

Reply:
I am changing it, I was unhappy with that.